// vite/monetizacion-app/src/ts/membresia.ts
export interface Membresia {
  id: number;
  artista: string;
  usuario: string;
  fechaInicio: string;
  activa: boolean;
}

export function crearFormularioMembresia() {
  const form = document.createElement('form');
  form.className = "bg-white p-4 rounded-lg shadow-md mb-6";

  form.innerHTML = `
    <h2 class="text-xl font-semibold mb-2">Registrar Membresía</h2>
    <input type="text" id="artista" placeholder="Artista" class="block mb-2 p-2 border rounded w-full"/>
    <input type="text" id="usuario" placeholder="Usuario" class="block mb-2 p-2 border rounded w-full"/>
    <input type="date" id="fechaInicio" class="block mb-2 p-2 border rounded w-full"/>
    <button type="submit" class="bg-blue-500 text-white px-4 py-2 rounded">Guardar</button>
    <div id="errores" class="text-red-500 mt-2"></div>
  `;

  form.addEventListener('submit', function (e) {
    e.preventDefault();

    const errores: string[] = [];
    const artista = (document.getElementById('artista') as HTMLInputElement).value.trim();
    const usuario = (document.getElementById('usuario') as HTMLInputElement).value.trim();
    const fechaInicio = (document.getElementById('fechaInicio') as HTMLInputElement).value;

    if (!artista) errores.push("El artista es obligatorio.");
    if (!usuario) errores.push("El usuario es obligatorio.");
    if (!fechaInicio) errores.push("La fecha de inicio es obligatoria.");

    const errorContainer = document.getElementById('errores')!;
    errorContainer.innerHTML = errores.map(err => `<p>${err}</p>`).join('');

    if (errores.length === 0) {
      errorContainer.innerHTML = "<p class='text-green-600'>¡Registro exitoso!</p>";
    }
  });

  document.body.appendChild(form);
}
